spec/src/main/asciidoc/repository.asciidoc

```diff
@@ -1,2 +1,217 @@
-[[repository]]
-== Repository
\ No newline at end of file
+
+= Repository
+
+In DDD, the repository pattern encapsulates the logic required to access data sources. They consolidate data access functionality, providing better maintainability and decoupling the infrastructure or technology used to access databases from the domain model layer.
+
+image::01-repository.png[alt=repository structure, width=70%, height=70%]
+
+This pattern focuses on the closest proximity of entities and hides where the data comes from.
+
+The Repository pattern is a well-documented way of working with a data source. In the book https://www.amazon.com/dp/0321127420/[Patterns of Enterprise Application] Architecture, Martin Fowler describes a repository as follows:
```

Comment:
Let's leave the Amazon links out of the Jakarta spec doc so that it doesn't appear that Jakarta is promoting a particular vendor.  The book and author name will be straightforward enough for anyone to copy into a search engine to find various vendors from which they can acquire it.

Reply:
Good point, thanks